pkg/nsx/services/nsxserviceaccount/cluster.go

```diff
@@ -186,12 +186,36 @@ func (s *NSXServiceAccountService) RestoreRealizedNSXServiceAccount(ctx context.
 	hasCCP := len(s.ClusterControlPlaneStore.GetByIndex(common.TagScopeNSXServiceAccountCRUID, string(obj.UID))) > 0
 	piObj := s.PrincipalIdentityStore.GetByKey(normalizedClusterName)
 	ccpObj := s.ClusterControlPlaneStore.GetByKey(normalizedClusterName)
+	var pi *mpmodel.PrincipalIdentity
+	var certificate mpmodel.Certificate
+	detail := true
+	if piObj != nil {
+		pi = piObj.(*mpmodel.PrincipalIdentity)
+		certificate, _ = s.NSXClient.CertificatesClient.Get(*(pi.CertificateId), &detail)
+	}
+	// read Secret
+	secretName := obj.Status.Secrets[0].Name
+	secretNamespace := obj.Status.Secrets[0].Namespace
+	secret := &v1.Secret{}
+	if err := s.Client.Get(ctx, types.NamespacedName{Name: secretName, Namespace: secretNamespace}, secret); err != nil {
+		return err
+	}
+	cert := secret.Data[SecretCertName]
+
+	log.V(1).Info("RestoreRealizedNSXServiceAccount: ",
+		"certificate.PemEncoded==nil", certificate.PemEncoded == nil,
+		"certificate.PemEncoded", certificate.PemEncoded,
+		"cert", string(cert))
 	if hasPI && hasCCP && piObj != nil && ccpObj != nil {
+		if string(cert) != "" && (certificate.PemEncoded == nil || certificate.PemEncoded != nil && *(certificate.PemEncoded) != string(cert)) {
```

Comment:
I think `(certificate.PemEncoded == nil || certificate.PemEncoded != nil && *(certificate.PemEncoded) != string(cert))` can be simplified a bit to:

```
(certificate.PemEncoded == nil || *(certificate.PemEncoded) != string(cert))
```

Reply:
done